tests/lib/rules/require-default-export.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * @author ItMaga
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const RuleTester = require('../../eslint-compat').RuleTester
+const rule = require('../../../lib/rules/require-default-export')
+
+const tester = new RuleTester({
+  languageOptions: {
+    parser: require('vue-eslint-parser'),
+    ecmaVersion: 2020,
+    sourceType: 'module'
+  }
+})
+
+tester.run('require-default-export', rule, {
+  valid: [
+    {
+      filename: 'test.vue',
+      code: `
+      <template>Without script</template>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+        import { ref } from 'vue';
+        
+        export default {}
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script setup>
+        const foo = 'foo';
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      const component = {};
+
+      export default component;
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      import {defineComponent} from 'vue';
+
+      export default defineComponent({});
+      </script>
+      `
+    },
+    {
+      filename: 'test.js',
+      code: `
+      const foo = 'foo';
+      export const bar = 'bar';
+      `
+    },
+    {
+      filename: 'test.js',
+      code: `
+      import {defineComponent} from 'vue';
+      defineComponent({});
+      `
+    }
+  ],
+  invalid: [
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      const foo = 'foo';
+      </script>
+      `,
+      errors: [
+        {
+          messageId: 'mustDefaultExport',
+          line: 1
```

Comment:
Can you change the reported range to be only the end `</script>` tag? That's usually where the default export goes. Also, please add `endLine`, `column` and `endColumn` to all test cases.